---
title: "Macrocystis WGS Project"
author: "Meaghan Clark"
date: "2026-02-24"
output: html_document
editor_options: 
  chunk_output_type: console
---
## Summary 

Project begun Februray 2026. Goal is to combine multiple *Macrocystis* whole genome resequencing data sets to learn about structure, inbreeding, and local adaptation. 


## Directory structure

```
MpyrAdapt/
├── data/
│   ├── raw/          # symbolic links to raw sequencing data
│   ├── processed/    # trimmed, filtered, aligned data
│   └── reference/    # symbolic link to reference genome
├── scripts/
│   ├── sra_download.sbatch
│   ├── read_QC.sbatch
│   └── 
├── 
│   ├── 
│   └── 
├── 
└── MpyrAdapt_notebook.md
```

## Metadata 

##### Site keys from Albert data 

---
AM	Andrew Molera	36.283333	-121.876392
MO	Monterey	36.617145	-121.895693
PL	Point Loma	32.678617	-117.253367
PP	Santa Cruz	36.952702	-121.9711
PV	Palos Verdes	33.720889	-118.336278
SC	Santa Cruz Island	34.032411	-119.524502
SM	San Miguel Island	34.051883	-120.33885
---

#### Gonzalez et al. PRJNA938791  
[Whole-genome sequencing distinguishes the two most common giant kelp ecomorphs](https://academic.oup.com/evolut/article/77/6/1354/7079449)

---
Site name	Site abbreviation	Coordinates	Hemisphere	Morph	Collected by
Stillwater Cove	STL	36.565058, -121.943553	North	Pyrifera and Integrifolia	S. T. Gonzalez
Cayucos	CAY	35.446456, -120.932394	North	Pyrifera	S. T. Gonzalez
Arroyo Quemado	AQ	34.47, -120.119	North	Pyrifera	F. Alberto
Catalina Island	CI	33.462, -118.511	North	Pyrifera	F. Alberto
Camp Pendleton	CB	33.157, -117.36	North	Pyrifera	F. Alberto
Playa Blanca	PLB	-28.1851111, -71.1651667	South	Integrifolia	S. T. Gonzalez
Punta Choros	PCH	-29.245712, -71.466174	South	Integrifolia	S. T. Gonzalez
Totorolillo	TOT	-32.022839, -71.508658	South	Integrifolia	S. T. Gonzalez
Las Docas	DOC	-33.139851, -71.708218	South	Integrifolia	S. T. Gonzalez
Algarrobo	ALG	-33.360022, -71.668728	South	Integrifolia	S. T. Gonzalez
Las Monjas	MON	-33.492179, -71.642485	South	Integrifolia	S. T. Gonzalez
La Boca	BOC	-33.903175, -71.834772	South	Pyrifera	S. T. Gonzalez
Tumbes	TUM	-36.623731, -73.093019	South	Pyrifera	S. T. Gonzalez
Chome	CHO	-36.806467, -73.177088	South	Pyrifera	S. T. Gonzalez
Bahía Mansa	BAM	-40.581556, -73.738197	South	Pyrifera	S. T. Gonzalez
Ancud	ANC	-41.865314, -73.831927	South	Pyrifera	S. T. Gonzalez
Dalcahue	DAL	-42.382756, -73.655023	South	Pyrifera	S. T. Gonzalez
---

#### Molano et al. PRJNA661280
[Sporophyte Stage Genes Exhibit Stronger Selection Than Gametophyte Stage Genes in Haplodiplontic Giant Kelp](https://www.frontiersin.org/journals/marine-science/articles/10.3389/fmars.2021.774076/full)

---
Sampling locations	Longitude	Latitude
Catalina Island	-118.511	33.462
Carlsbad	-117.36	33.157
Santa Barbara/Arroyo Quemada	-120.119	34.47
---

#### Download genomes from SRA 

install SRA toolkit to download data from the SRA to elkhorn
```{bash, eval = FALSE}
# install SRA toolkit 
# Download the latest Linux binary
# from /home/miclark/software/
wget https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-ubuntu64.tar.gz
tar -xzf sratoolkit.current-ubuntu64.tar.gz

# Add to PATH
echo 'export PATH=$HOME/software/sratoolkit.3.3.0-ubuntu64/bin:$PATH' >> ~/.bashrc
source ~/.bashrc

# configure SRA toolkit
# follow guidance from: https://github.com/ncbi/sra-tools/wiki/05.-Toolkit-Configuration
vdb-config --interactive
  # in cache, set user-repository to /hb/scratch/miclark/sra

```

Download SRA data
```{bash, eval = FALSE}
sbatch ./scripts/sra_download.sbatch PRJNA938791 # Submitted batch job 106781
    # "SRR023" is text from the URL that hosts the sequences, included in SRR_accessions accidently 
sbatch ./scripts/sra_download.sbatch PRJNA661280 # Submitted batch job 106787
    # "SRR012" is text from the URL that hosts the sequences, included in SRR_accessions accidently 

# [] remove sra files and unzipped .fastq files 
rm -r Mpyr_WGS_data/PRJNA*/sra/
rm Mpyr_WGS_data/PRJNA*/fastq/*.fastq

```

## Raw data processing 

Standardize file names and set up symbolic links from Mpyr_WGS_data to MpyrAdapt/data/raw
```{bash, eval = FALSE}

# standardize file names 


# set up symbolic links from Mpyr_WGS_data to MpyrAdapt/data/raw


```

Run FastQC on raw reads, run MultiQC to summarize results, and check percent adapter dimer
```{bash, eval = FALSE}
sbatch read_QC.sbatch "<full path to directory with the files to be processed>" "<report name>"
sbatch "./scripts/read_QC.sbatch data/raw" "fastq_raw"

```

Trim raw reads with Fastp
```{bash, eval = FALSE}

```





