---
title: "Macrocystis WGS Project"
author: "Meaghan Clark"
date: "2026-02-24"
output: html_document
editor_options: 
  chunk_output_type: console
---
## Summary 

Project begun Februray 2026. Goal is to combine multiple *Macrocystis* whole genome resequencing data sets to learn about structure, inbreeding, and local adaptation. 


## Directory structure

```
MpyrAdapt/
├── data/
│   ├── raw/          # symbolic links to raw sequencing data
│   ├── processed/    # trimmed, filtered, aligned data
│   └── reference/    # symbolic link to reference genome
├── scripts/
│   ├── sra_download.sbatch
│   ├── read_QC.sbatch
│   └── 
├── 
│   ├── 
│   └── 
├── 
└── MpyrAdapt_notebook.md
```

## Metadata 

##### Site keys from Alberto data 

---
AM	Andrew Molera	36.283333	-121.876392
MO	Monterey	36.617145	-121.895693
PL	Point Loma	32.678617	-117.253367
PP	Santa Cruz	36.952702	-121.9711
PV	Palos Verdes	33.720889	-118.336278
SC	Santa Cruz Island	34.032411	-119.524502
SM	San Miguel Island	34.051883	-120.33885
---

#### Gonzalez et al. PRJNA938791  
[Whole-genome sequencing distinguishes the two most common giant kelp ecomorphs](https://academic.oup.com/evolut/article/77/6/1354/7079449)

---
Site name	Site abbreviation	Coordinates	Hemisphere	Morph	Collected by
Stillwater Cove	STL	36.565058, -121.943553	North	Pyrifera and Integrifolia	S. T. Gonzalez
Cayucos	CAY	35.446456, -120.932394	North	Pyrifera	S. T. Gonzalez
Arroyo Quemado	AQ	34.47, -120.119	North	Pyrifera	F. Alberto
Catalina Island	CI	33.462, -118.511	North	Pyrifera	F. Alberto
Camp Pendleton	CB	33.157, -117.36	North	Pyrifera	F. Alberto
Playa Blanca	PLB	-28.1851111, -71.1651667	South	Integrifolia	S. T. Gonzalez
Punta Choros	PCH	-29.245712, -71.466174	South	Integrifolia	S. T. Gonzalez
Totorolillo	TOT	-32.022839, -71.508658	South	Integrifolia	S. T. Gonzalez
Las Docas	DOC	-33.139851, -71.708218	South	Integrifolia	S. T. Gonzalez
Algarrobo	ALG	-33.360022, -71.668728	South	Integrifolia	S. T. Gonzalez
Las Monjas	MON	-33.492179, -71.642485	South	Integrifolia	S. T. Gonzalez
La Boca	BOC	-33.903175, -71.834772	South	Pyrifera	S. T. Gonzalez
Tumbes	TUM	-36.623731, -73.093019	South	Pyrifera	S. T. Gonzalez
Chome	CHO	-36.806467, -73.177088	South	Pyrifera	S. T. Gonzalez
Bahía Mansa	BAM	-40.581556, -73.738197	South	Pyrifera	S. T. Gonzalez
Ancud	ANC	-41.865314, -73.831927	South	Pyrifera	S. T. Gonzalez
Dalcahue	DAL	-42.382756, -73.655023	South	Pyrifera	S. T. Gonzalez
---

#### Molano et al. PRJNA661280
[Sporophyte Stage Genes Exhibit Stronger Selection Than Gametophyte Stage Genes in Haplodiplontic Giant Kelp](https://www.frontiersin.org/journals/marine-science/articles/10.3389/fmars.2021.774076/full)

---
Sampling locations	Longitude	Latitude
Catalina Island	-118.511	33.462
Carlsbad	-117.36	33.157
Santa Barbara/Arroyo Quemada	-120.119	34.47
---

#### Download genomes from SRA 

install SRA toolkit to download data from the SRA to elkhorn
```{bash, eval = FALSE}
# install SRA toolkit 
# Download the latest Linux binary
# from /home/miclark/software/
wget https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-ubuntu64.tar.gz
tar -xzf sratoolkit.current-ubuntu64.tar.gz

# Add to PATH
echo 'export PATH=$HOME/software/sratoolkit.3.3.0-ubuntu64/bin:$PATH' >> ~/.bashrc
source ~/.bashrc

# configure SRA toolkit
# follow guidance from: https://github.com/ncbi/sra-tools/wiki/05.-Toolkit-Configuration
vdb-config --interactive
  # in cache, set user-repository to /hb/scratch/miclark/sra

```

Download SRA data
```{bash, eval = FALSE}
sbatch ./scripts/sra_download.sbatch PRJNA938791 # Submitted batch job 106781
    # "SRR023" is text from the URL that hosts the sequences, included in SRR_accessions accidently 
sbatch ./scripts/sra_download.sbatch PRJNA661280 # Submitted batch job 106787
    # "SRR012" is text from the URL that hosts the sequences, included in SRR_accessions accidently 

# compression is taking a long time -- make parallel in an interactive job
  # cancelling when compressing SRR23817627_1.fastq and SRR12582529_2.fastq 
salloc --time=24:00:00 --mem=100Gb --ntasks=1 --cpus-per-task=32 --partition=lab-mpinsky \
        --qos=pi-mpinsky \
        --account=pi-mpinsky

# PRJNA661280
INDIR=/home/miclark/Mpyr_WGS_data/PRJNA661280/fastq
module load parallel/20200122
ls ${INDIR}/*.fastq | parallel --no-notice -j 32 "gzip {}"
rm /home/miclark/Mpyr_WGS_data/PRJNA661280/fastq/SRR12582529_2.fastq.gz
gzip /home/miclark/Mpyr_WGS_data/PRJNA661280/fastq/SRR12582529_2.fastq

# PRJNA938791 [x]
INDIR=/home/miclark/Mpyr_WGS_data/PRJNA938791/fastq
rm /home/miclark/Mpyr_WGS_data/PRJNA938791/fastq/SRR23817627_1.fastq.gz # [x] 
ls ${INDIR}/*.fastq | parallel --no-notice -j 32 "gzip {}" # [x]

# moved sra files to scratch 

# generate md5sums 
cd /home/miclark/Mpyr_WGS_data/PRJNA938791/fastq/
md5sum ./*.fastq.gz > ./md5sums.txt
cd /home/miclark/Mpyr_WGS_data/PRJNA661280/fastq/
md5sum ./*.fastq.gz > ./md5sums.txt
cd /home/miclark/Mpyr_WGS_data/Mpyr_Alberto_data
md5sum ./*.fastq.gz > ./md5sums.txt

# move fastq.gz files to scratch 
mv /home/miclark/Mpyr_WGS_data/* /data/rufus/pinsky_lab/Mpyr/non_GCRG_WGS/ #[x]

# verify md5sums
cd /data/rufus/pinsky_lab/Mpyr/non_GCRG_WGS/PRJNA938791/fastq/
md5sum -c ./md5sums.txt > ./md5sums_check.txt # [x]
cd /data/rufus/pinsky_lab/Mpyr/non_GCRG_WGS/PRJNA661280/fastq/
md5sum -c ./md5sums.txt > ./md5sums_check.txt # [x]
cd /data/rufus/pinsky_lab/Mpyr/non_GCRG_WGS/Mpyr_Alberto_data
md5sum -c ./md5sums.txt > ./md5sums_check.txt # []

# get run information, which includes individual sample names, saves to runinfo.csv in each project directory
cd /data/rufus/pinsky_lab/Mpyr/non_GCRG_WGS/PRJNA938791/
wget -q "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=sra&id=$(grep -o '"[0-9]\{6,\}"' ./search.json | tr -d '"' | paste -sd',')&rettype=runinfo&retmode=csv" -O runinfo.csv
cd /data/rufus/pinsky_lab/Mpyr/non_GCRG_WGS/PRJNA661280/
wget -q "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=sra&id=$(grep -o '"[0-9]\{6,\}"' ./search.json | tr -d '"' | paste -sd',')&rettype=runinfo&retmode=csv" -O runinfo.csv


```

## Raw data processing 

Standardize file names and set up symbolic links from Mpyr_WGS_data to MpyrAdapt/data/raw
  - Naming convetion: Species_project_site-sample_number_readDirection, e.g. Mpyr_ALB_STL-03_1, Mpyr_MOL_STL-03_2, Mpyr_GON_STL-03_1

```{bash, eval = FALSE}

# set up symbolic links from Mpyr_WGS_data to MpyrAdapt/data/raw
sbatch ./scripts/make_symbolic_links.sbatch # Submitted batch job 109551

# two files not found
    # WARNING: /data/rufus/pinsky_lab/Mpyr/non_GCRG_WGS/PRJNA938791/fastq/SRR23817635_1.fastq.gz not found, skipping
    # WARNING: /data/rufus/pinsky_lab/Mpyr/non_GCRG_WGS/PRJNA938791/fastq/SRR23817635_2.fastq.gz not found, skipping
    # SRR23817635 is a sequence that exists from Gonzalez et al, but is missing from sra download. This is a southern hemisphere integrifolia sample, which is likely outside the scope of this project, so we will proceed without it 

# set up symbolic link to reference genome
ln -s /hb/groups/mpinsky_lab/Mpyr_ref_2025/hap1/Mpyr-NLJ1B.v3.hap1.softmasked.fasta /home/miclark/MpyrAdapt/data/reference/Mpyr-NLJ1B.v3.hap1.softmasked.fasta

```

Run FastQC on raw reads, run MultiQC to summarize results, and check percent adapter dimer
```{bash, eval = FALSE}
sbatch ./scripts/read_QC.sbatch "/home/miclark/MpyrAdapt/data/raw" "raw_reads_QC" # Submitted batch job 109557

```

Trim raw reads with Fastp
```{bash, eval = FALSE}
sbatch ./scripts/trim_reads.sbatch

```





